### ISSUE TITLE
Ho to limit the range of the super-resolved image

### ISSUE THREAD

**Issue by antonelse** [XPixelGroup/HAT#132]
Hi, just a curiosity.

What are the main parameters that influence the final output range if I do not consider the final scaling and rescaling thus the plain `x` [here](https://github.com/XPixelGroup/HAT/blob/1b22ba0aff82d9d041f5bfa763f82649e6c23d99/hat/archs/hat_arch.py#L971), without `x = (x - self.mean) * self.img_range` and `x = x / self.img_range + self.mean`.

Because I'm trying to use it with an image bounded between **0 and 1**, but it gives output values from **-30 to +200**, and I don't know why. I'm doing this because I want to apply this algorithm to non-real images, like temperature or gas emissions maps.

If i force the net to output values bounded between 0 and 1, i.e. `x[x<0]=0` and `x[x>1]=1` I obtain images filled with ones, thus the training stops after some epochs since the loss is nearly constant and does not improve.

Thanks a lot!

**Comment by chxy95** [comment]
@antonelse The model process data in float32 and the image is processed into [0,1] range in the data loader. If you want to save the float data instead of the uint8 data ranging [0,255], modify the model file avoiding **tensor2img**. https://github.com/XPixelGroup/BasicSR/blob/033cd6896d898fdd3dcda32e3102a792efa1b8f4/basicsr/models/sr_model.py#L208

**Comment by antonelse** [comment]
Thanks @chxy95 !
The problem is that, after some batches, the network tries to shrink the values to match the input range of the image, but after some epochs, it explodes, returning values far above the original range. 
To avoid this, I wanted to know how I could act to reduce the problem in the image synthesis part and prevent this unusual behavior. 
I'll experiment with some regularization and gradient clipping techniques.
If you have any suggestions, I'm here :)
Many thanks.
